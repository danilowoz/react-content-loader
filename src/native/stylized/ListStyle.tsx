import * as React from 'react'

import Holder, { Rect } from '../Holder'
import { IContentLoaderProps } from '../'

const ReactContentLoaderListStyle: React.FC<IContentLoaderProps> = props => (
  <Holder {...props}>
    <Rect x="0" y="0" rx="3" ry="3" width="250" height="10" />
    <Rect x="20" y="20" rx="3" ry="3" width="220" height="10" />
    <Rect x="20" y="40" rx="3" ry="3" width="170" height="10" />
    <Rect x="0" y="60" rx="3" ry="3" width="250" height="10" />
    <Rect x="20" y="80" rx="3" ry="3" width="200" height="10" />
    <Rect x="20" y="100" rx="3" ry="3" width="80" height="10" />
  </Holder>
)

export default ReactContentLoaderListStyle
