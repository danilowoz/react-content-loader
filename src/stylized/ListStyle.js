import React from 'react'
import Wrap from '../Wrap'

const ListStyle = (props) => {
    return (
        <Wrap {...props}>
            <rect x="0" y="0" rx="3" ry="3" width="250" height="10" />
            <rect x="20" y="20" rx="3" ry="3" width="220" height="10" />
            <rect x="20" y="40" rx="3" ry="3" width="170" height="10" />
            <rect x="0" y="60" rx="3" ry="3" width="250" height="10" />
            <rect x="20" y="80" rx="3" ry="3" width="200" height="10" />
            <rect x="20" y="100" rx="3" ry="3" width="80" height="10" />
        </Wrap>
    )
}

export default ListStyle
