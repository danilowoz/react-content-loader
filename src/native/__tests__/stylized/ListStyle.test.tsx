import 'react-native'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import ListStyle from '../../stylized/ListStyle'

describe('ListStyle', () => {
  const wrapper = renderer.create(
    <ListStyle id="ListStyle" animate={false} speed={20} />
  )

  test('renders correctly', () => {
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('props are propagated ', () => {
    expect(wrapper.root.props.speed).toEqual(20)
  })
})
