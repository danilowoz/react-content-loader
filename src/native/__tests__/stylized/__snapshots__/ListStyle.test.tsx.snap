// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListStyle renders correctly 1`] = `
<Svg
  height={130}
  preserveAspectRatio="none"
  style={Object {}}
  viewBox="0 0 400 130"
  width={400}
>
  <Rect
    clipPath="url(#ListStyle-idGradient)"
    fill="url(#ListStyle-idClip)"
    height={130}
    width={400}
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="ListStyle-idGradient"
    >
      <Rect
        height="10"
        rx="3"
        ry="3"
        width="250"
        x="0"
        y="0"
      />
      <Rect
        height="10"
        rx="3"
        ry="3"
        width="220"
        x="20"
        y="20"
      />
      <Rect
        height="10"
        rx="3"
        ry="3"
        width="170"
        x="20"
        y="40"
      />
      <Rect
        height="10"
        rx="3"
        ry="3"
        width="250"
        x="0"
        y="60"
      />
      <Rect
        height="10"
        rx="3"
        ry="3"
        width="200"
        x="20"
        y="80"
      />
      <Rect
        height="10"
        rx="3"
        ry="3"
        width="80"
        x="20"
        y="100"
      />
    </ClipPath>
    <LinearGradient
      id="ListStyle-idClip"
      x1="-100%"
      x2="100%"
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f0f0f0"
      />
      <Stop
        offset={0}
        stopColor="#e0e0e0"
      />
      <Stop
        offset={0}
        stopColor="#f0f0f0"
      />
    </LinearGradient>
  </Defs>
</Svg>
`;
